### ISSUE TITLE
`InitVar` + `ClassVar` inside `dataclass` causes wrong `__init__` generation

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3159]
### Summary

Not sure why you would ever do this, but if you do ty gets confused and makes an incorrect `__init__`
[Playground](https://play.ty.dev/50860f01-40f2-4f90-94f8-d39d84a875b7)
```py
from typing import reveal_type, Final, ClassVar
from dataclasses import dataclass, InitVar

@dataclass
class A:
    x: InitVar[ClassVar[int]]

# Runtime: TypeError: A.__init__() missing 1 required positional argument: 'x'
reveal_type(A())
```
The fun part is that because of how `dataclass` works, if you swap the order to `ClassVar[InitVar[int]]` ty is now correct, since the `InitVar` checking only checks the top level. Also see https://docs.python.org/3/library/dataclasses.html#init-only-variables

### Version

ty 0.0.26 (940305127 2026-03-26) playground 229e49f20

**Comment by carljm** [comment]
Mypy and pyright emit a diagnostic anytime `ClassVar` is nested inside another type or qualifier. I suspect nesting either `ClassVar` or `InitVar` should be an error? Both make sense only at top level. (Maybe with an exception for inside `Annotated`? Pyright seems to make that exception; mypy does not.
